src/plzinit/plugins.go

```diff
@@ -0,0 +1,285 @@
+package plzinit
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/please-build/gcfg/ast"
+	"github.com/thought-machine/please/src/core"
+	"github.com/thought-machine/please/src/plz"
+)
+
+const pluginRepoTemplate = `plugin_repo(
+  name = "%s",
+  revision = "%s",
+  plugin = "%s-rules",
+  owner = "please-build",
+)
+`
+
+// InitPlugins initialises one or more plugins by inserting plugin config values into
+// the host repo config file, and creating a build target in //plugins.
+func InitPlugins(plugins []string) {
+	log.Debug("Initialising plugin(s): %v", plugins)
+	for _, p := range plugins {
+		if err := initPlugin(p); err != nil {
+			log.Warningf("could not initialise plugin %s. Got error: %v", p, err)
+		}
+	}
+}
+
+func initPlugin(plugin string) error {
+	if err := createTarget("plugins/BUILD", plugin); err != nil {
+		return err
+	}
+	if err := injectPluginConfig(plugin); err != nil {
+		return err
+	}
+	return nil
+}
+
+func injectPluginConfig(plugin string) error {
+	configPath := path.Join(core.RepoRoot, ".plzconfig")
+	if _, err := os.Stat(configPath); errors.Is(err, os.ErrNotExist) {
+		log.Warningf("config file %s does not exist", configPath)
+		return err
+	}
+
+	configFile, err := os.Open(configPath)
+	if err != nil {
+		return err
+	}
+	defer configFile.Close()
+	file := ast.Read(configFile)
+	switch plugin {
+	case "python":
+		file = writePythonConfigFields(file)
+	case "java":
+		file = writeJavaConfigFields(file)
+	case "cc":
+		file = writeCCConfigFields(file)
+	default:
+		log.Fatalf("failed to initialise plugin. \"%s\" not recognised", plugin)
```

Comment:
I wonder if this should really fail here. What `injectPluginConfig` is trying to do is actually migrating old config to new config. There will be new plugins that won't require any migration eventually and so the inject logic should just be skipped.